src/main/java/org/jabref/gui/push/PushToApplicationsManager.java

```diff
@@ -27,13 +27,13 @@ public PushToApplicationsManager(DialogService dialogService) {
 
     public static PushToApplicationSettings getSettings(PushToApplication application) {
```

Comment:
If you convert this to an instance method and define it in  the  `ExternalTab` constructor you can then init it with the dialogService and can directly access the dialogService method.

     manager = new PushToApplicationsManager(dialogService);
So you don't need the getDialogService method in

Reply:
Yes, I thought about this, but the problem is: the PushToApplicationManager is created in the JabRefFrame - as it is. The PushToApplicationButton in the toolbar is depending on this, so I can't move it in ExternalTabs, so it would probably have to be done on this very upmost level. But as there are other managers in the JabRefFrame used the same way, I am cautious breaking the consistency of the code.
Next thing is: If I create an object of this, I screw up all that other stuff about the "applications"-stuff in the manager etc.

Reply:
or did i missunderstand you. Probably. got to think

Reply:
ok, I tried to  simplify it, got rid of these stupid getDialogService calls. I really don't know why i put them there in the first place.
So the logic keeps the same, the dialogService is now handed over from ExternalTab through the static method to the Settings-Constructor.